sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,460 @@
+    >>> trigsimp(lower_bob.frame.dcm(ceiling.frame))
+    Matrix([
+    [ cos(theta_P1(t) + theta_P2(t)), sin(theta_P1(t) + theta_P2(t)), 0],
+    [-sin(theta_P1(t) + theta_P2(t)), cos(theta_P1(t) + theta_P2(t)), 0],
+    [                              0,                              0, 1]])
+
+    The position of the lower bob's masscenter is found with:
+
+    >>> lower_bob.masscenter.pos_from(ceiling.masscenter)
+    l1*U_frame.x + l2*L_frame.x
+
+    The angular velocities of the two pendulum links can be computed with
+    respect to the ceiling.
+
+    >>> upper_bob.frame.ang_vel_in(ceiling.frame)
+    omega_P1(t)*C_frame.z
+    >>> lower_bob.frame.ang_vel_in(ceiling.frame)
+    omega_P1(t)*C_frame.z + omega_P2(t)*U_frame.z
+
+    And finally, the linear velocities of the two pendulum bobs can be computed
+    with respect to the ceiling.
+
+    >>> upper_bob.masscenter.vel(ceiling.frame)
+    l1*omega_P1(t)*U_frame.y
+    >>> lower_bob.masscenter.vel(ceiling.frame)
+    l1*omega_P1(t)*U_frame.y + l2*(omega_P1(t) + omega_P2(t))*L_frame.y
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None,
+                 parent_joint_pos=None, child_joint_pos=None, parent_axis=None,
+                 child_axis=None):
+
+        super().__init__(name, parent, child, coordinates, speeds,
+                         parent_joint_pos, child_joint_pos, parent_axis,
+                         child_axis)
+
+    def __str__(self):
+        return (f'PinJoint: {self.name}  parent: {self.parent}  '
+                f'child: {self.child}')
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            theta = dynamicsymbols('theta' + '_' + self._name)
+            coordinate = theta
+        coordinates.append(coordinate)
+        return coordinates
+
+    def _generate_speeds(self, speed):
+        speeds = []
+        if speed is None:
+            omega = dynamicsymbols('omega' + '_' + self._name)
+            speed = omega
+        speeds.append(speed)
+        return speeds
+
+    def _orient_frames(self):
+        self.child.frame.orient_axis(self.parent.frame, self.parent_axis, 0)
+        angle, axis = self._alignment_rotation(self.parent_axis,
+                                               self.child_axis)
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=UserWarning)
+            if axis != Vector(0):
+
+                int_frame = ReferenceFrame('int_frame')
+                int_frame.orient_axis(self.child.frame, self.child_axis, 0)
+                int_frame.orient_axis(self.parent.frame, axis, angle)
+                self.child.frame.orient_axis(int_frame, self.parent_axis,
+                                            self.coordinates[0])
+            else:
+                self.child.frame.orient_axis(self.parent.frame, self.parent_axis,
+                                            self.coordinates[0])
```

Comment:
For the case when axis == 0, the angle is an integer multiple of pi. For the cases pi, 3pi, 5pi, etc how does this ensure that the rotation of child wrt to parent is about an axis pointing opposite in each frame? I'm not seeing that. This seems to do the same thing for the 0 degree and 180 degree cases.